tortoise/algorithm.go

```diff
@@ -120,6 +120,11 @@ func (t *Tortoise) OnBeacon(eid types.EpochID, beacon types.Beacon) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
 	evicted := t.trtl.evicted.GetEpoch()
+	// when the node recover from checkpoint, the genesis epoch and recover epoch can be the same
+	if types.GetEffectiveGenesis() != types.FirstEffectiveGenesis() &&
+		evicted == (types.GetEffectiveGenesis()+1).GetEpoch() {
+		evicted--
```

Comment:
this doesn't look good

is it possible to initialize tortoise state correctly? right now t.evicted is set to genesis - 1. what it needs to be set so that it will work both for recovery and normal case?